server/ladder_service.py

```diff
@@ -164,6 +169,42 @@ async def fetch_matchmaker_queues(self, conn):
             ))
         return matchmaker_queues
 
+    def parse_map_params(self, map_list: List[Union[Map, NeroxisGeneratedMap]], map_params: str, weight: int):
+        try:
```

Comment:
I would put the `try` block around the call, and I'd just return the `NeroxisGeneratedMap` here, and append it to the list where it's called.